### PR TITLE
Add `deepsparse.transformers.eval_downstream` for BERT

### PR DESCRIPTION
**Pull request by mgoin**
Allows evaluating trained BERT models on downstream datasets: squad, mnli, qqp, sst2

```
deepsparse.transformers.eval_downstream zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/pruned_quant_3layers-aggressive_89 --dataset squad

100%|█████████████████████████████████████████████████████████████████████| 10570/10570 [03:49<00:00, 46.02it/s]

SQuAD eval results: {'exact_match': 69.10122989593188, 'f1': 78.35950221732533}
```

### CODE REVIEW

File: src/deepsparse/transformers/eval_downstream.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+        type=int,
+        default=None,
+    )
+
+    return parser.parse_args()
+
+
+def squad_eval(args):
+    # load squad validation dataset and eval tool
+    squad = load_dataset("squad")["validation"]
```
Comment (rahul-tuli) [inline src/deepsparse/transformers/eval_downstream.py:132]:
```suggestion
    squad = load_dataset(SQUAD_KEY)["validation"]
```

File: src/deepsparse/transformers/eval_downstream.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    )
+    parser.add_argument(
+        "onnx_filepath",
+        type=str,
+        help="The full filepath of the ONNX model file or SparseZoo stub to the model",
+    )
+    parser.add_argument(
+        "-d",
+        "--dataset",
+        type=str,
```
Comment (rahul-tuli) [inline src/deepsparse/transformers/eval_downstream.py:89]: If dataset is not specified we get the following error:
```
Traceback (most recent call last):
  File "/home/rahul/projects/deepsparse/src/deepsparse/transformers/eval_downstream.py", line 274, in <module>
    main()
  File "/home/rahul/projects/deepsparse/src/deepsparse/transformers/eval_downstream.py", line 257, in main
    args.dataset = args.dataset.lower()
AttributeError: 'NoneType' object has no attribute 'lower'

Process finished with exit code 1
```

Suggested FIx:
```suggestion
        type=str,
        required=True,
```

File: src/deepsparse/transformers/eval_downstream.py [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+        mnli_eval(args)
+    elif args.dataset == QQP_KEY:
+        qqp_eval(args)
+    elif args.dataset == SST2_KEY:
+        sst2_eval(args)
+    else:
+        raise KeyError(
+            f"Unknown downstream dataset {args.dataset}, "
+            f"available datasets are {SUPPORTED_DATASETS}"
+        )
```
Comment (rahul-tuli) [inline src/deepsparse/transformers/eval_downstream.py:279]: Can be removed, `argparse` more specifically line 90 takes care of this for us if invalid dataset provided,

```
eval_downstream.py: error: argument -d/--dataset: invalid choice: 's' (choose from 'squad', 'mnli', 'qqp', 'sst2')
```

```suggestion
```

File: src/deepsparse/transformers/eval_downstream.py [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+
+    args.dataset = args.dataset.lower()
+    if args.dataset == SQUAD_KEY:
+        squad_eval(args)
+    elif args.dataset == MNLI_KEY:
+        mnli_eval(args)
+    elif args.dataset == QQP_KEY:
+        qqp_eval(args)
+    elif args.dataset == SST2_KEY:
+        sst2_eval(args)
```
Comment (rahul-tuli) [inline src/deepsparse/transformers/eval_downstream.py:274]: Would a dictionary mapping of keys to function be cleaner? thoughts on that? I do not dislike this approach either,

**Review by rahul-tuli — CHANGES_REQUESTED** [review]
LGTM! minor change requested for `--dataset` argument

**Comment by rahul-tuli** [comment]
I like how clean the code is.

File: src/deepsparse/transformers/eval_downstream.py (hunk 4 above)
Comment (mgoin) [reply src/deepsparse/transformers/eval_downstream.py:275]: yeah that would work, just need to reorder the functions above the argparse then

**Review by rahul-tuli — APPROVED** [review]

**Pull request merged**